### PR TITLE
fix: outfits saving/deleting

### LINKED ISSUE #8849: [QA] Outfits | Invisible or deleted outfits when saving outfits quickly
### 🛠️ **Build version:**
149

### 📔 **Issue Description:**
it seems that outfits are broken when we attempt to save or delete multiples of them in the same time

### 👣 **STR:** 

  1. Open app
  2. Erase all outfits quickly 
  3. Save outfits on the 5 row or 10 quickly 
  4. Change the panel to another one (could be map , communities and more )
  5. return to outfits 
  6. Check there are missing outfits on the outfit section 

### ✔️ **Expected Result:**
No outfit should be missing 

### ❌ **Actual Result with evidence:**
Invisible or deleted outfits when saving outfits quickly 

##### ⚠️ Intermittent - (note)

https://github.com/user-attachments/assets/7360b047-ecd8-4d93-956d-a4f9cef568ec

#### 🖥️ **Operative system and additional Notes:**
This happens in both platforms

### PR DESCRIPTION
**Pull request by biotech77**
# Pull Request Description
Closes #8849 and #5849 

## What does this PR change?
Simplifies the process of saving/deleting outfits by allowing only one at the time to be saved or deleted.
Other improvements:
- while saving outfit animation will be stopped in order to save the thumbnail in default pose (previously it could be saved during the animation)
- thumbnail will be preserved in memory and will be saved only if saving (solves edge case where we can start saving outfit but then equip another one causing wrong thumbnail to be saved)

### Test Steps
It's possible to save or delete only one outfit at the time. You can still equip while save/delete operation is ongoing

1. Save outfit
2. Observe that you can't save another one or delete until previous operation is complete

1. Delete outfit
2. Observe that you can't delete another one or save until previous operation is complete

### Additional Testing Notes
- Note any edge cases to verify
- Mention specific areas that need careful testing
- List known limitations or potential issues

## Quality Checklist
- [x] Changes have been tested locally
- [ ] Documentation has been updated (if required)
- [ ] Performance impact has been considered
- [ ] For SDK features: Test scene is included

## Code Review Reference
Please review our [Branch & PR Standards](../docs/branch-and-pr-standards.md) before submitting. It explains the automated review flow, QA/DEV approval requirements, and what each label does — especially useful for first-time contributors.

### CODE REVIEW

**Comment by m3taphysics** [comment]
PR [#8900](https://github.com/decentraland/unity-explorer/pull/8900), run [#26978043303](https://github.com/decentraland/performance-testing/actions/runs/26978043303)

Builds: [Windows change](https://explorer-artifacts.decentraland.org/@dcl/unity-explorer/branch/fix/8849-outfits-saving-deleting/pr-22754-5e835ac/Decentraland_windows64.zip), [Windows baseline](https://explorer-artifacts.decentraland.org/@dcl/unity-explorer/releases/v0.150.0-alpha/Decentraland_windows64.zip), [macOS change](https://explorer-artifacts.decentraland.org/@dcl/unity-explorer/branch/fix/8849-outfits-saving-deleting/pr-22754-5e835ac/Decentraland_macos.zip), [macOS baseline](https://explorer-artifacts.decentraland.org/@dcl/unity-explorer/releases/v0.150.0-alpha/Decentraland_macos.zip)

<details><summary>How to read this table</summary>

- Each build is measured **3 times**. The values are the **median**, and **(min–max)** is the lowest and highest of those runs — a wide range means the metric is noisy and small differences are not trustworthy.
- **Δ** is Change minus Baseline (a negative Δ means Change is faster).
- **🟢 faster / 🔴 slower** — a real difference: larger than **both** 3% **and** the run-to-run range.
- **⚪ within noise** — the difference is smaller than how much the build varies between its own runs, so it cannot be told apart from random variation. Treat it as no change.
- **Exceptions logged** — exceptions found in the run logs; more than the baseline is flagged 🔴 even when frame times look fine.
</details>

### Framework 13 i7

| Metric | Baseline | Change | Δ | Result |
| --- | ---: | ---: | ---: | :--- |
| Samples | 2700 (×3) | 2701 (×3) | | |
| CPU average | 33.3 ms | 33.3 ms | -0.0 ms | ⚪ within noise |
| CPU 1% worst | 34.2 ms (33.8–34.5) | 33.8 ms (33.6–33.8) | -0.4 ms | ⚪ within noise |
| CPU 0.1% worst | 39.2 ms (36.4–44.8) | 35.7 ms (35.1–36.5) | -3.5 ms | ⚪ within noise |
| GPU average | 7.2 ms (7.1–7.5) | 8.2 ms (8.2–8.3) | 1.0 ms | 🔴 14% slower |
| GPU 1% worst | 19.5 ms (18.6–20.0) | 16.5 ms (16.3–17.0) | -3.0 ms | 🟢 15% faster |
| GPU 0.1% worst | 25.0 ms (24.0–25.8) | 18.5 ms (18.3–21.1) | -6.5 ms | 🟢 26% faster |
| Exceptions logged | 46 | 42 | -4 | 🟢 fewer errors |

**Comment by m3taphysics** [comment]
PR [#8900](https://github.com/decentraland/unity-explorer/pull/8900), run [#27017957770](https://github.com/decentraland/performance-testing/actions/runs/27017957770)

Builds: [Windows change](https://explorer-artifacts.decentraland.org/@dcl/unity-explorer/branch/fix/8849-outfits-saving-deleting/pr-22768-b883009/Decentraland_windows64.zip), [Windows baseline](https://explorer-artifacts.decentraland.org/@dcl/unity-explorer/releases/v0.150.0-alpha/Decentraland_windows64.zip), [macOS change](https://explorer-artifacts.decentraland.org/@dcl/unity-explorer/branch/fix/8849-outfits-saving-deleting/pr-22768-b883009/Decentraland_macos.zip), [macOS baseline](https://explorer-artifacts.decentraland.org/@dcl/unity-explorer/releases/v0.150.0-alpha/Decentraland_macos.zip)

<details><summary>How to read this table</summary>

- Each build is measured **3 times**. The values are the **median**, and **(min–max)** is the lowest and highest of those runs — a wide range means the metric is noisy and small differences are not trustworthy.
- **Δ** is Change minus Baseline (a negative Δ means Change is faster).
- **🟢 faster / 🔴 slower** — a real difference: larger than **both** 3% **and** the run-to-run range.
- **⚪ within noise** — the difference is smaller than how much the build varies between its own runs, so it cannot be told apart from random variation. Treat it as no change.
- **Exceptions logged** — exceptions found in the run logs; more than the baseline is flagged 🔴 even when frame times look fine.
</details>

### Framework 13 i7

| Metric | Baseline | Change | Δ | Result |
| --- | ---: | ---: | ---: | :--- |
| Samples | 2701 (×3) | 2701 (×3) | | |
| CPU average | 33.3 ms | 33.3 ms | -0.0 ms | ⚪ within noise |
| CPU 1% worst | 34.4 ms (33.9–34.7) | 33.7 ms (33.6–34.0) | -0.7 ms | ⚪ within noise |
| CPU 0.1% worst | 43.1 ms (36.5–45.7) | 36.0 ms (35.7–36.6) | -7.2 ms | ⚪ within noise |
| GPU average | 7.3 ms (7.3–7.7) | 8.5 ms (8.4–8.8) | 1.2 ms | 🔴 16% slower |
| GPU 1% worst | 19.0 ms (18.5–21.0) | 18.5 ms (17.9–18.5) | -0.6 ms | ⚪ within noise |
| GPU 0.1% worst | 24.8 ms (24.0–26.3) | 25.7 ms (24.0–25.8) | 0.9 ms | ⚪ within noise |
| Exceptions logged | 42 | 42 | 0 | ⚪ none new |

**Comment by m3taphysics** [comment]
PR [#8900](https://github.com/decentraland/unity-explorer/pull/8900), run [#27126154683](https://github.com/decentraland/performance-testing/actions/runs/27126154683)

Builds: [Windows change](https://explorer-artifacts.decentraland.org/@dcl/unity-explorer/branch/fix/8849-outfits-saving-deleting/pr-22782-12bef69/Decentraland_windows64.zip), [Windows baseline](https://explorer-artifacts.decentraland.org/@dcl/unity-explorer/releases/v0.150.0-alpha/Decentraland_windows64.zip), [macOS change](https://explorer-artifacts.decentraland.org/@dcl/unity-explorer/branch/fix/8849-outfits-saving-deleting/pr-22782-12bef69/Decentraland_macos.zip), [macOS baseline](https://explorer-artifacts.decentraland.org/@dcl/unity-explorer/releases/v0.150.0-alpha/Decentraland_macos.zip)

<details><summary>How to read this table</summary>

- Each build is measured **3 times**. The values are the **median**, and **(min–max)** is the lowest and highest of those runs — a wide range means the metric is noisy and small differences are not trustworthy.
- **Δ** is Change minus Baseline (a negative Δ means Change is faster).
- **🟢 faster / 🔴 slower** — a real difference: larger than **both** 3% **and** the run-to-run range.
- **⚪ within noise** — the difference is smaller than how much the build varies between its own runs, so it cannot be told apart from random variation. Treat it as no change.
- **Exceptions logged** — exceptions found in the run logs; more than the baseline is flagged 🔴 even when frame times look fine.
</details>

### Framework 13 i7

| Metric | Baseline | Change | Δ | Result |
| --- | ---: | ---: | ---: | :--- |
| Samples | 2701 (×3) | 2701 (×3) | | |
| CPU average | 33.3 ms | 33.3 ms | -0.0 ms | ⚪ within noise |
| CPU 1% worst | 33.9 ms (33.7–34.5) | 33.6 ms (33.4–33.9) | -0.3 ms | ⚪ within noise |
| CPU 0.1% worst | 37.5 ms (35.8–43.1) | 36.4 ms (33.9–39.8) | -1.1 ms | ⚪ within noise |
| GPU average | 7.2 ms (7.0–8.2) | 8.2 ms (8.1–8.3) | 1.0 ms | ⚪ within noise |
| GPU 1% worst | 19.3 ms (17.5–19.9) | 18.0 ms (17.8–18.6) | -1.3 ms | ⚪ within noise |
| GPU 0.1% worst | 24.6 ms (23.0–24.6) | 24.2 ms (24.1–26.6) | -0.3 ms | ⚪ within noise |
| Exceptions logged | 48 | 42 | -6 | 🟢 fewer errors |

**Review by Ludmilafantaniella — APPROVED** [review]
Tested on Windows and Mac. ✅

- Saving an outfit blocks further save/delete actions until the operation completes
- Deleting an outfit blocks further delete/save actions until the operation completes
- Equipping another outfit mid-save does not cause wrong thumbnail to be saved
- Thumbnails are saved correctly in default pose
- No missing or invisible outfits after rapid save/delete sequences

All flows working correctly. LGTM 🚀

https://github.com/user-attachments/assets/b46df20f-1b97-40f6-bf9b-5ba4d6804614

https://github.com/user-attachments/assets/d4afbca0-b8f5-4f6c-a0b6-66a3b95ce778

✅Smoke test performed:
- ✔️ Backpack and wearables in world
- ✔️ Emotes in world and in backpack
- ✔️ Teleport with map/coordinates/Jump In
- ✔️ Profile card
- ✔️ Camera
- ✔️ Skybox

**Comment by m3taphysics** [comment]
PR [#8900](https://github.com/decentraland/unity-explorer/pull/8900), run [#27192526002](https://github.com/decentraland/performance-testing/actions/runs/27192526002)

Builds: [Windows change](https://explorer-artifacts.decentraland.org/@dcl/unity-explorer/branch/fix/8849-outfits-saving-deleting/pr-22814-7ae2a16/Decentraland_windows64.zip), [Windows baseline](https://explorer-artifacts.decentraland.org/@dcl/unity-explorer/releases/v0.150.0-alpha/Decentraland_windows64.zip), [macOS change](https://explorer-artifacts.decentraland.org/@dcl/unity-explorer/branch/fix/8849-outfits-saving-deleting/pr-22814-7ae2a16/Decentraland_macos.zip), [macOS baseline](https://explorer-artifacts.decentraland.org/@dcl/unity-explorer/releases/v0.150.0-alpha/Decentraland_macos.zip)

<details><summary>How to read this table</summary>

- Each build is measured **3 times**. The values are the **median**, and **(min–max)** is the lowest and highest of those runs — a wide range means the metric is noisy and small differences are not trustworthy.
- **Δ** is Change minus Baseline (a negative Δ means Change is faster).
- **🟢 faster / 🔴 slower** — a real difference: larger than **both** 3% **and** the run-to-run range.
- **⚪ within noise** — the difference is smaller than how much the build varies between its own runs, so it cannot be told apart from random variation. Treat it as no change.
- **Exceptions logged** — exceptions found in the run logs; more than the baseline is flagged 🔴 even when frame times look fine.
</details>

### Framework 13 i7

| Metric | Baseline | Change | Δ | Result |
| --- | ---: | ---: | ---: | :--- |
| Samples | 2700 (×3) | 2701 (×3) | | |
| CPU average | 33.3 ms | 33.3 ms | -0.0 ms | ⚪ within noise |
| CPU 1% worst | 34.2 ms (33.8–34.3) | 33.4 ms (33.4–33.6) | -0.8 ms | ⚪ within noise |
| CPU 0.1% worst | 41.3 ms (35.8–42.2) | 33.5 ms (33.5–35.2) | -7.8 ms | 🟢 19% faster |
| GPU average | 7.3 ms (7.3–7.7) | 8.1 ms (8.0–8.7) | 0.8 ms | 🔴 10% slower |
| GPU 1% worst | 19.5 ms (18.6–19.7) | 18.7 ms (18.2–18.8) | -0.8 ms | ⚪ within noise |
| GPU 0.1% worst | 23.5 ms (23.4–25.4) | 25.2 ms (23.8–26.0) | 1.7 ms | ⚪ within noise |
| Exceptions logged | 48 | 12 | -36 | 🟢 fewer errors |

**Comment by m3taphysics** [comment]
PR [#8900](https://github.com/decentraland/unity-explorer/pull/8900), run [#27353625290](https://github.com/decentraland/performance-testing/actions/runs/27353625290)

Builds: [Windows change](https://explorer-artifacts.decentraland.org/@dcl/unity-explorer/branch/fix/8849-outfits-saving-deleting/pr-22938-42368e9/Decentraland_windows64.zip), [Windows baseline](https://explorer-artifacts.decentraland.org/@dcl/unity-explorer/releases/v0.151.0-alpha/Decentraland_windows64.zip), [macOS change](https://explorer-artifacts.decentraland.org/@dcl/unity-explorer/branch/fix/8849-outfits-saving-deleting/pr-22938-42368e9/Decentraland_macos.zip), [macOS baseline](https://explorer-artifacts.decentraland.org/@dcl/unity-explorer/releases/v0.151.0-alpha/Decentraland_macos.zip)

<details><summary>How to read this table</summary>

- Each build is measured **3 times**. The values are the **median**, and **(min–max)** is the lowest and highest of those runs — a wide range means the metric is noisy and small differences are not trustworthy.
- **Δ** is Change minus Baseline (a negative Δ means Change is faster).
- **🟢 faster / 🔴 slower** — a real difference: larger than **both** 3% **and** the run-to-run range.
- **⚪ within noise** — the difference is smaller than how much the build varies between its own runs, so it cannot be told apart from random variation. Treat it as no change.
- **Exceptions logged** — exceptions found in the run logs; more than the baseline is flagged 🔴 even when frame times look fine.
</details>

### Framework 13 i7

| Metric | Baseline | Change | Δ | Result |
| --- | ---: | ---: | ---: | :--- |
| Samples | 2701 (×3) | 2701 (×3) | | |
| CPU average | 33.3 ms | 33.3 ms | -0.0 ms | ⚪ within noise |
| CPU 1% worst | 33.8 ms (33.7–34.0) | 33.5 ms (33.4–33.7) | -0.3 ms | ⚪ within noise |
| CPU 0.1% worst | 37.1 ms (36.0–37.2) | 34.1 ms (33.6–36.9) | -3.0 ms | ⚪ within noise |
| GPU average | 7.5 ms (7.5–7.6) | 7.7 ms (7.6–7.9) | 0.2 ms | ⚪ within noise |
| GPU 1% worst | 19.7 ms (19.2–20.0) | 18.8 ms (18.2–20.1) | -0.8 ms | ⚪ within noise |
| GPU 0.1% worst | 24.8 ms (24.1–26.8) | 24.1 ms (23.6–25.9) | -0.7 ms | ⚪ within noise |
| Exceptions logged | 12 | 12 | 0 | ⚪ none new |

**Comment by m3taphysics** [comment]
PR [#8900](https://github.com/decentraland/unity-explorer/pull/8900), run [#27433617952](https://github.com/decentraland/performance-testing/actions/runs/27433617952)

Builds: [Windows change](https://explorer-artifacts.decentraland.org/@dcl/unity-explorer/branch/fix/8849-outfits-saving-deleting/pr-23048-e53982e/Decentraland_windows64.zip), [Windows baseline](https://explorer-artifacts.decentraland.org/@dcl/unity-explorer/releases/v0.151.0-alpha/Decentraland_windows64.zip), [macOS change](https://explorer-artifacts.decentraland.org/@dcl/unity-explorer/branch/fix/8849-outfits-saving-deleting/pr-23048-e53982e/Decentraland_macos.zip), [macOS baseline](https://explorer-artifacts.decentraland.org/@dcl/unity-explorer/releases/v0.151.0-alpha/Decentraland_macos.zip)

<details><summary>How to read this table</summary>

- Each build is measured **3 times**. The values are the **median**, and **(min–max)** is the lowest and highest of those runs — a wide range means the metric is noisy and small differences are not trustworthy.
- **Δ** is Change minus Baseline (a negative Δ means Change is faster).
- **🟢 faster / 🔴 slower** — a real difference: larger than **both** 3% **and** the run-to-run range.
- **⚪ within noise** — the difference is smaller than how much the build varies between its own runs, so it cannot be told apart from random variation. Treat it as no change.
- **Exceptions per run** — the average number of exceptions in a run's log; more than the baseline is flagged 🔴 even when frame times look fine. The *Exception breakdown* under each table groups them by the explorer's report category and exception type (as totals across the runs).
- A run that logged **unusually many exceptions** (at least 10 and 5× the median of its build's runs — e.g. a service was down during it) is excluded from all numbers and called out under the table.
</details>

### Framework 13 i7

| Metric | Baseline | Change | Δ | Result |
| --- | ---: | ---: | ---: | :--- |
| Samples | 2701 (×3) | 2701 (×3) | | |
| CPU average | 33.3 ms | 33.3 ms | 0.0 ms | ⚪ within noise |
| CPU 1% worst | 33.8 ms (33.7–34.0) | 33.7 ms (33.7–33.9) | -0.0 ms | ⚪ within noise |
| CPU 0.1% worst | 37.1 ms (36.0–37.2) | 35.8 ms (35.5–36.0) | -1.3 ms | 🟢 4% faster |
| GPU average | 7.5 ms (7.5–7.6) | 7.8 ms (7.6–8.0) | 0.3 ms | ⚪ within noise |
| GPU 1% worst | 19.7 ms (19.2–20.0) | 18.8 ms (18.0–20.1) | -0.9 ms | ⚪ within noise |
| GPU 0.1% worst | 24.8 ms (24.1–26.8) | 23.4 ms (23.0–29.6) | -1.3 ms | ⚪ within noise |
| Exceptions per run | 2 | 3 | +1 | 🔴 more errors |

<details><summary>Exception breakdown</summary>

| Exception | Baseline (3 runs) | Change (3 runs) |
| --- | ---: | ---: |
| `[ENGINE] NullReferenceException` | 6 | 6 |
| `[ENGINE] ObjectDisposedException` | 0 | 3 |
</details>

**Comment by m3taphysics** [comment]
⚠️ Performance test [run](https://github.com/decentraland/performance-testing/actions/runs/27436726532) cancelled: **Framework 13 i7 / Test** still had not started after 60 minutes — the runner is offline or stuck. Re-run when the machine is back, or remove it from the ENABLED_RUNNERS variable.

File: Explorer/Assets/DCL/Backpack/AvatarSection/Outfits/Repository/OutfitsRepository.cs [hunk 1]
```diff
@@ -20,128 +18,167 @@ namespace DCL.Backpack.AvatarSection.Outfits.Repository
...
             { Converters = new List<JsonConverter> { new ColorJsonConverter() } };
 
-        private const int DEFAULT_DEPLOY_WINDOW_IN_SECONDS = 15;
+        /// <summary>
+        ///     Minimum number of seconds we wait between successive publishes to the
+        ///     Catalyst content endpoint. This matches the backend's per-pointer rate limit on
+        ///     <c>:outfits</c> entity submissions — publishing inside the cooldown gets rejected.
+        ///     Configurable in production via the <c>OUTFITS_DEPLOY_WINDOW</c> feature flag.
+        /// </summary>
+        private const int DEFAULT_DEPLOY_COOLDOWN_SECONDS = 15;
```
Comment (lorenzo-ranciaffi) [inline Explorer/Assets/DCL/Backpack/AvatarSection/Outfits/Repository/OutfitsRepository.cs:34]: We have different windows depending on what we are pushing to the catalyst. Are we sure about the window length and if it's not shared with the profile? If they are, we'll get 400s as the gateings we are doing are completely independent from eachother

File: Explorer/Assets/DCL/Backpack/AvatarSection/Outfits/OutfitsPresenter.cs [hunk 2]
```diff
@@ -182,62 +183,123 @@ private async UniTaskVoid OnSaveOutfitRequestedAsync(int slotIndex, Cancellation
...
+                if (capture == null)
+                {
+                    RevertSlot(presenter, originalOutfitData);
+                    return;
+                }
+
+                // Optimistic display: show the thumbnail immediately so the user gets feedback
+                // during the ~15s deploy window. PNG is only persisted to disk after the
+                // backend save succeeds (PersistPngAsync below).
+                presenter.SetThumbnail(capture.Value.Thumbnail);
```
Comment (lorenzo-ranciaffi) [inline Explorer/Assets/DCL/Backpack/AvatarSection/Outfits/OutfitsPresenter.cs:225]: Is there anything telling the user the outfit is saving? If they have to wait the full 15s there's a high chance the panel will get closed -> flow cancelled -> no outfit saved even though it showed the thumbnail

**Review by lorenzo-ranciaffi — CHANGES_REQUESTED** [review]
Good job! I just have a couple of concerns

**Comment by m3taphysics** [comment]
⚠️ Performance test [run](https://github.com/decentraland/performance-testing/actions/runs/27532844706) cancelled: **Framework 13 i7 / Test** still had not started after 60 minutes — the runner is offline or stuck. Re-run when the machine is back, or remove it from the ENABLED_RUNNERS variable.

**Comment by m3taphysics** [comment]
⚠️ Performance test [run](https://github.com/decentraland/performance-testing/actions/runs/27536038167) cancelled: **Framework 13 i7 / Test** still had not started after 60 minutes — the runner is offline or stuck. Re-run when the machine is back, or remove it from the ENABLED_RUNNERS variable.

File: Explorer/Assets/DCL/Backpack/AvatarSection/Outfits/OutfitsPresenter.cs (hunk 2 above)
Comment (biotech77) [reply Explorer/Assets/DCL/Backpack/AvatarSection/Outfits/OutfitsPresenter.cs:225]: @lorenzo-ranciaffi yes, we have a progress spinner, screenshot is captured and kept until successful deployment, only then is put into slot and saved on the disk, before it could happen that you can equip other outfit in meanwhile and thumbnail is recorded afterwards capturing the wrong one. Also now animations are stopped so we don't capture it with animation.

File: Explorer/Assets/DCL/Backpack/AvatarSection/Outfits/Repository/OutfitsRepository.cs (hunk 1 above)
Comment (biotech77) [reply Explorer/Assets/DCL/Backpack/AvatarSection/Outfits/Repository/OutfitsRepository.cs:34]: They're not shared with the profile: the catalyst keys its deployment rate limiter per entity type and per pointer ([src/logic/deployment-service/rate-limiter.ts](https://github.com/decentraland/catalyst/blob/main/src/logic/deployment-service/rate-limiter.ts)), and profile (type=profile, pointer <wallet>) vs outfits (type=outfits, pointer <wallet>:outfits) land in completely separate caches.

15s is default but it's overridden with feature flag and the current value is 4s

**Review by lorenzo-ranciaffi — APPROVED** [review]
